tests/System.Text.Http.Parser.Tests/HttpParserTests.cs

```diff
@@ -373,8 +373,8 @@ public void ParseRequestLineSplitBufferWithoutNewLineDoesNotUpdateConsumed()
             Assert.Equal(1, pairs.Length);
             Assert.Equal(headerName, pairs[0].Key);
             Assert.Equal(expectedHeaderValue, pairs[0].Value);
-            Assert.Equal(buffer.End, consumed);
```

Comment:
This is an interesting side effect of the change: Positions returned from GetPosition do not have type information in them (we need 2 bits for it) so cannot be directly compared to other cursors (Start/End)